qutip/tests/core/test_qobjevo.py

```diff
@@ -65,51 +92,52 @@ def _cplx(t, args):
     return np.exp(1j*t*args['w2'])
 
 
-@pytest.fixture(params=[
-    pytest.param("cte", id="cte"),
-    pytest.param("func", id="func"),
-    pytest.param("string", id="string"),
-    pytest.param("spline", id="spline"),
-    pytest.param("array", id="array"),
-    pytest.param("array_log", id="array_log"),
-    pytest.param("with_args", id="with_args"),
-    pytest.param("no_args", id="no_args"),
-    pytest.param("mixed_tlist", id="mixed_tlist"),
-    pytest.param("qobjevofunc", id="qobjevofunc"),
-])
-def form(request):
-    return request.param
+real_qevo = Pseudo_qevo(
+    rand_stochastic(N).to(data.CSR),
+    rand_stochastic(N).to(data.CSR),
+    _real, "sin(t*w1)", args)
 
+herm_qevo = Pseudo_qevo(
+    rand_herm(N).to(data.Dense),
+    rand_herm(N).to(data.Dense),
+    _real, "sin(t*w1)", args)
 
-@pytest.fixture(params=[
-    pytest.param("func", id="func"),
-    pytest.param("string", id="string"),
-    pytest.param("with_args", id="with_args"),
-    pytest.param("qobjevofunc", id="qobjevofunc"),
-])
-def args_form(request):
+cplx_qevo = Pseudo_qevo(
+    rand_stochastic(N).to(data.Dense),
+    rand_stochastic(N).to(data.CSR) + rand_stochastic(N).to(data.CSR) * 1j,
+    _cplx, "exp(1j*t*w2)", args)
+
+
+@pytest.fixture(params=['func_coeff', 'string', 'spline',
+                        'array', 'logarray', 'func_call'])
+def coeff_type(request):
+    # all available QobjEvo types
     return request.param
 
 
 @pytest.fixture(params=[
-    pytest.param("cte", id="cte"),
-    pytest.param("func", id="func"),
-    pytest.param("string", id="string"),
-    pytest.param("spline", id="spline"),
-    pytest.param("array", id="array"),
-    pytest.param("array_log", id="array_log"),
-    pytest.param("with_args", id="with_args"),
-    pytest.param("no_args", id="no_args"),
-    pytest.param("mixed_tlist", id="mixed_tlist"),
-    pytest.param("qobjevofunc", id="qobjevofunc"),
+    pytest.param(real_qevo, id="real"),
+    pytest.param(herm_qevo, id="hermitian"),
+    pytest.param(cplx_qevo, id="complex"),
 ])
-def extra_form(request):
+def pseudo_qevo(request):
     return request.param
```

Comment:
Not sure if you intended this, but you'll use the same three random pseudo-QobjEvo objects all the way through - the randomness only happens once.

Reply:
Yes this is intended. Recreating the Qobj would only take time without any effect on the validity of the tests.